unison-cli/src/Unison/LSP.hs

```diff
@@ -58,12 +58,18 @@ spawnLsp codebase runtime latestBranch latestPath =
     UnliftIO.handleIO (handleFailure lspPort) $ do
       TCP.serve (TCP.Host "127.0.0.1") lspPort $ \(sock, _sockaddr) -> do
         Ki.scoped \scope -> do
-          sockHandle <- socketToHandle sock ReadWriteMode
+          let clientInput = do
+                -- The server will be in the process of shutting down if the socket is closed,
+                -- so just return empty input in the meantime.
+                fromMaybe "" <$> TCP.recv sock defaultChunkSize
```

Comment:
In theory this could lead to an infinite loop of trying to get more data from the client because it never returns any new bytes, but in reality `TCP.serve` will always shut down this thread when the socket closes so that'll never come up.